build/templates/_converters.py.mako

```diff
@@ -164,55 +223,137 @@ def test_convert_timedelta_to_microseconds_int():
+    test_result, test_result_list = convert_repeated_capabilities(['0', '1'], prefix='ScriptTrigger')
+    assert test_result_list == ['ScriptTrigger0', 'ScriptTrigger1']
     assert test_result == 'ScriptTrigger0,ScriptTrigger1'
-    test_result = convert_repeated_capabilities([0, 1], prefix='ScriptTrigger')
+    test_result, test_result_list = convert_repeated_capabilities([0, 1], prefix='ScriptTrigger')
+    assert test_result_list == ['ScriptTrigger0', 'ScriptTrigger1']
+    assert test_result == 'ScriptTrigger0,ScriptTrigger1'
+
+
+def test_repeated_capabilies_tuple_channel():
+    test_result, test_result_list = convert_repeated_capabilities(('0'))
+    assert test_result_list == ['0']
+    assert test_result == '0'
+    test_result, test_result_list = convert_repeated_capabilities(('0,1'))
+    assert test_result_list == ['0', '1']
+    assert test_result == '0,1'
+    test_result, test_result_list = convert_repeated_capabilities(('0', '1'))
+    assert test_result_list == ['0', '1']
+    assert test_result == '0,1'
+    test_result, test_result_list = convert_repeated_capabilities((0, 1))
+    assert test_result_list == ['0', '1']
+    assert test_result == '0,1'
+    test_result, test_result_list = convert_repeated_capabilities((0, 1, '3'))
+    assert test_result_list == ['0', '1', '3']
+    assert test_result == '0,1,3'
+
+
+def test_repeated_capabilies_tuple_prefix():
+    test_result, test_result_list = convert_repeated_capabilities(('ScriptTrigger0,ScriptTrigger1'), prefix='ScriptTrigger')
+    assert test_result_list == ['ScriptTrigger0', 'ScriptTrigger1']
+    assert test_result == 'ScriptTrigger0,ScriptTrigger1'
+    test_result, test_result_list = convert_repeated_capabilities(('0'), prefix='ScriptTrigger')
+    assert test_result_list == ['ScriptTrigger0']
+    assert test_result == 'ScriptTrigger0'
+    test_result, test_result_list = convert_repeated_capabilities(('0', '1'), prefix='ScriptTrigger')
+    assert test_result_list == ['ScriptTrigger0', 'ScriptTrigger1']
+    assert test_result == 'ScriptTrigger0,ScriptTrigger1'
+    test_result, test_result_list = convert_repeated_capabilities((0, 1), prefix='ScriptTrigger')
+    assert test_result_list == ['ScriptTrigger0', 'ScriptTrigger1']
     assert test_result == 'ScriptTrigger0,ScriptTrigger1'
 
 
 def test_repeated_capabilies_slice_channel():
-    test_result = convert_repeated_capabilities(slice(0, 1))
+    test_result, test_result_list = convert_repeated_capabilities(slice(0, 1))
+    assert test_result_list == ['0']
     assert test_result == '0'
-    test_result = convert_repeated_capabilities(slice(0, 2))
+    test_result, test_result_list = convert_repeated_capabilities(slice(0, 2))
+    assert test_result_list == ['0', '1']
     assert test_result == '0,1'
-    test_result = convert_repeated_capabilities(slice(None, 2))
+    test_result, test_result_list = convert_repeated_capabilities(slice(None, 2))
+    assert test_result_list == ['0', '1']
```

Comment:
In all these tests, it's weird that you test the second thing returned by the function first.

Reply:
Only returning the list so only test that.